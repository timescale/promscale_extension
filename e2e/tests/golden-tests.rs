use crate::common::{exec_sql_script, run_postgres};
use insta::assert_snapshot;
use std::env;
use test_generator::test_resources;
use testcontainers::clients;

mod common;

#[test_resources("testdata/*.sql")]
fn golden_test(resource: &str) {
    let docker = clients::Cli::default();
    let node = run_postgres(&docker);
    let query_result = exec_sql_script(&node, resource);

    assert_snapshot!(resource, query_result);
}
