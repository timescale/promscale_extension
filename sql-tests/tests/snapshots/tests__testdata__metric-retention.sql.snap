---
source: e2e/tests/golden-tests.rs
expression: query_result
---
 plan 
------
 1..4
(1 row)

                        is                         
---------------------------------------------------
 ok 1 - default metric retention period is 90 days
(1 row)

 get_or_create_metric_table_name 
---------------------------------
 (1,cpu_usage,t)
(1 row)

 set_metric_retention_period 
-----------------------------
 t
(1 row)

                             is                              
-------------------------------------------------------------
 ok 2 - get_metric_retention_period returns retention period
(1 row)

                             is                              
-------------------------------------------------------------
 ok 3 - get_metric_retention_period returns retention period
(1 row)

 set_default_retention_period 
------------------------------
 t
(1 row)

                                 is                                  
---------------------------------------------------------------------
 ok 4 - get_default_metric_retention_period returns retention period
(1 row)

 finish 
--------
(0 rows)


